[Gutenberg] Hide help button from UBE

SiobhyB opened the pull request:
Addresses https://github.com/wordpress-mobile/gutenberg-mobile/issues/4339

## Related PRs and Installable Builds

* `Gutenberg`: https://github.com/WordPress/gutenberg/pull/37221
* `Gutenberg Mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4352
* `WPAndroid`: https://github.com/wordpress-mobile/WordPress-Android/pull/15674

## Description

As described in https://github.com/wordpress-mobile/gutenberg-mobile/issues/4339, a help button is currently displaying in the bottom right-hand corner of the unsupported block editor (UBE) in both the Android and iOS apps. The button can be seen while logged into WordPress.com. When tapped on, the button has no impact on iOS.

With this PR, the button is hidden from view in order to avoid confusion and potential loss of edits. A high-level overview of the code changes can be found in the `Types of changes` section of [the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/37221).

## Testing

Please refer to [the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/37221) as the "central PR" with the most up-to-date testing instructions. 

<hr />

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

SiobhyB committed:
Update reference to Mobile Gutenberg

SiobhyB committed:
Enable custom CSS styles for Gutenberg's UBE

In order to allow custom, external CSS styles, the pattern set for allowing custom JS scripts has been followed i.e. a function similar to 'onGutenbergReadyScripts()' has been created, 'onGutenbergReadyStyles()', along with a function for processing the styles, 'loadCustomStyles()'.

SiobhyB committed:
Merge branch 'develop' into gutenberg/update/hide-help-button-from-ube

SiobhyB committed:
Update reference to Gutenberg Mobile

SiobhyB committed:
Update reference to Gutenberg Mobile

SiobhyB committed:
Update project.pbxproj to include new .css file

SiobhyB committed:
Update reference to Gutenberg Mobile

SiobhyB committed:
Update reference to Gutenberg Mobile

twstokes commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergWeb/GutenbergWebViewController.swift:
```diff
@@ -82,6 +82,12 @@ class GutenbergWebViewController: GutenbergWebSingleBlockViewController, WebKitA
         ].compactMap { $0 }
     }
 
+    override func onGutenbergReadyStyles() -> [WKUserScript] {
```
Just wanted to flag [this comment from the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/37221#discussion_r771089838) about the function name.

twstokes commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergWeb/GutenbergWebViewController.swift:
```diff
@@ -100,6 +106,16 @@ class GutenbergWebViewController: GutenbergWebSingleBlockViewController, WebKitA
         }
     }
 
+    private func loadCustomStyles(named name: String, with argument: String? = nil) -> WKUserScript? {
```
I've [proposed an idea on the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/37221#discussion_r771093813) about how we may be able to simplify this a little, if you wouldn't mind looking at it first. 🙇

twstokes replied:
If we keep this function the way it is we may want to drop `with argument: String? = nil` since it isn't currently being used.

SiobhyB replied:
I've gone ahead to update this as per your suggestion in https://github.com/wordpress-mobile/WordPress-iOS/pull/17644/commits/df2615cd5bd97a55fd44612dcf6548a7a32dda2e :)

twstokes commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergWeb/GutenbergWebViewController.swift:
```diff
@@ -82,6 +82,12 @@ class GutenbergWebViewController: GutenbergWebSingleBlockViewController, WebKitA
         ].compactMap { $0 }
     }
 
+    override func onGutenbergReadyStyles() -> [WKUserScript] {
```
Just wanted to flag [this comment from the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/37221#discussion_r771089838) about the function name.

SiobhyB replied:
Thank you, changed in https://github.com/wordpress-mobile/WordPress-iOS/pull/17644/commits/9a2cf3b0ef36555a72a911a0d3a446187741b062. Let me know if you have further thoughts on the naming, though!

SiobhyB committed:
Rename function to 'onGutenbergLoadStyles' for clarity

SiobhyB committed:
Simplify loadCustomStyles function

Following the refactoring of the SourceFile function (as outlined here:
https://github.com/WordPress/gutenberg/pull/37221/commits/0cda8b3e569faba16bc10aa39c60d44658e92268), the loadCustomStyles function can now be simplified to just call SourceFile() directly.

SiobhyB committed:
Merge branch 'develop' into gutenberg/update/hide-help-button-from-ube

SiobhyB committed:
Update reference to Mobile Gutenberg

twstokes reviewed (approved):
LGTM @SiobhyB to merge after pointing to the appropriate tag. 👍

SiobhyB committed:
Merge branch 'develop' into gutenberg/update/hide-help-button-from-ube

SiobhyB committed:
Update reference to Mobile Gutenberg
